File: STRINGS/1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.scala
object Solution {
  def maxDepth(s: String): Int = {
    var st = List[Int]()
    var maxCounter = 0
    
    for (char <- s) {
      if (char == '(') {
        st = 1 :: st
      } else if (char == ')') {
        st = st.tail
      }
      maxCounter = math.max(maxCounter, st.length)
    }
    
    maxCounter
  }
}

File: STRINGS/1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        Stack<Integer> st = new Stack<>();
        int maxCounter = 0;
        for (int i = 0; i < s.length(); ++i) {
            if (s.charAt(i) == '(') {
                st.push(i);
            } else if (s.charAt(i) == ')') {
                st.pop();
            }
            maxCounter = Math.max(maxCounter, st.size());
        }
        return maxCounter;
    }
}